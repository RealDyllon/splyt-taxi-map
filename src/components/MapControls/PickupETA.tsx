import React from 'react';
import styled from 'styled-components';

interface Props {
  pickupETA: number;
}

const Wrapper = styled.div`
  position: absolute;
  top: 16px;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 999;
`;

const Container = styled.div`
  background: #ffffff;
  height: 50px;
  border-radius: 25px;
  padding: 0 32px;
  display: flex;
  justify-content: center;
  align-items: center;
  box-shadow: rgba(17, 17, 26, 0.1) 0px 1px 0px,
    rgba(17, 17, 26, 0.1) 0px 8px 24px,
    rgba(17, 17, 26, 0.1) 0px 16px 48px;
`;

const PickupETA = (props: Props) => {
  const { pickupETA } = props;

  return (
    <Wrapper>
      <Container>
        Ride ETA:&nbsp;<b>{pickupETA} mins</b>
      </Container>
    </Wrapper>
  );
};

export default PickupETA;
