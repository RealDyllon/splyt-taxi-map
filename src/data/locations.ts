import { LatLngTuple } from 'leaflet';

export interface locationType {
  id: number;
  name: string;
  coords: LatLngTuple;
}

const singapore: locationType = {
  id: 1,
  name: 'Singapore',
  coords: [1.285194, 103.8522982],
};

const london: locationType = {
  id: 2,
  name: 'London',
  coords: [51.5049375, -0.0964509],
};

const locations = {
  singapore,
  london,
};

export default locations;
