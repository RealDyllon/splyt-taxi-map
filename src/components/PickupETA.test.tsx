import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import locations from '../data/locations';
import driversApiRes from '../data/mock/driversApi';
import PickupETA from './PickupETA';

const TestComponent = <PickupETA pickupETA={5} />;

it('renders without crashing', () => {
  shallow(TestComponent);
});

it('renders correctly', () => {
  const tree = shallow(TestComponent);
  expect(toJson(tree)).toMatchSnapshot();
});
