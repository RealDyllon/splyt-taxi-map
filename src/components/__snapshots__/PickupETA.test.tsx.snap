// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    Ride ETA:Â 
     
    <b>
      5
       mins
    </b>
  </styled.div>
</styled.div>
`;
